src/LifecycleService.ts

```diff
@@ -76,14 +77,23 @@ export interface LifecycleService {
 
 }
 
+interface ClientForLifecycleService {
+    getLoggingService(): LoggingService;
```

Comment:
We can pass `LoggingService` directly. What's the point in calling getter on the client object? My suggestion was only for  non-trivial (non-getter) methods like `doShutdown()` where we have circular behavior dependencies between the client and underlying services.

Also, in `doShutdown()` particular case we can move the control into the client itself (see https://github.com/hazelcast/hazelcast-nodejs-client/pull/797#discussion_r572054545). I mean something like the following:
```ts
    shutdown(): Promise<void> {
        // we call a hook here now
        if (!this.lifecycleService.onShutdownStart()) {
            return Promise.resolve();
        }

        if (this.mapRepairingTask !== undefined) {
            this.mapRepairingTask.shutdown();
        }
        this.nearCacheManager.destroyAllNearCaches();
        this.proxyManager.destroy();
        this.statistics.stop();
        return this.cpSubsystem.shutdown()
            .then(() => {
                this.connectionManager.shutdown();
                this.invocationService.shutdown();
                // and another hook here
                this.lifecycleService.onShutdownFinished();
            });
    }
```

Reply:
For now, I did not change shutdown process of client. It can be done later

Reply:
I'd prefer to include such changes into this PR as it's aimed to refactor code.

Reply:
Ok I will do it, but I need to understand the logic behind hooks you showed. You can correct me in reviews